File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/domain/QuestionType.java
package io.zenwave360.example.clinicaltool.modules.surveys.domain;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.util.Arrays;

/**
 * Enum for QuestionType.
 */
public enum QuestionType {
    
    YES_NO(1),
    MULTIPLE_SELECTION(2),
    SINGLE_SELECTION(3),
    TEXT(4),
    NUMBER(5),
    DECIMAL(6),
    RANGE(7),
    ;
    private final Integer value;

    private QuestionType(Integer value) {
        this.value = value;
    }

    public Integer getValue() {
        return value;
    }

    public static QuestionType fromValue(Integer value) {
        return Arrays.stream(QuestionType.values())
                .filter(e -> e.value.equals(value)).findFirst().orElse(null);
    }
    @Converter
    static class QuestionTypeConverter implements AttributeConverter<QuestionType, Integer> {

        @Override
        public Integer convertToDatabaseColumn(QuestionType attribute) {
            if (attribute == null) {
                return null;
            }

            return attribute.value;
        }

        @Override
        public QuestionType convertToEntityAttribute(Integer dbData) {
            return QuestionType.fromValue(dbData);
        }

    }
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/example/asyncapi/provider/clinicaltool/modules/surveys/domain/QuestionType.kt
package io.example.asyncapi.provider.clinicaltool.modules.surveys.domain
    import jakarta.persistence.AttributeConverter
    import jakarta.persistence.Converter

/**
* Enum for QuestionType.
*/
enum class QuestionType(val value: Int) {

    YES_NO(1),
    MULTIPLE_SELECTION(2),
    SINGLE_SELECTION(3),
    TEXT(4),
    NUMBER(5),
    DECIMAL(6),
    RANGE(7)
;
    companion object {
    fun fromValue(value: Int): QuestionType? {
    return values().find { it.value == value }
    }
    }
        @Converter
        class QuestionTypeConverter : AttributeConverter<QuestionType, Int> {
        override fun convertToDatabaseColumn(attribute: QuestionType?): Int? {
        return attribute?.value
        }

        override fun convertToEntityAttribute(dbData: Int?): QuestionType? {
        return if (dbData == null) null else QuestionType.fromValue(dbData)
        }
        }
}

